packages/builder/src/components/automation/SetupPanel/RowSelector.svelte

```diff
@@ -1,61 +1,168 @@
-  import { TableNames } from "constants"
+  import { FIELDS } from "constants/backend"
+  import { capitalise } from "helpers"
+  import { memo } from "@budibase/frontend-core"
+  import PropField from "./PropField.svelte"
+  import { cloneDeep, isPlainObject, mergeWith } from "lodash"
 
   const dispatch = createEventDispatcher()
-  export let value
+
+  export let row
   export let meta
   export let bindings
   export let isTestModal
-  export let isUpdateRow
 
-  $: parsedBindings = bindings.map(binding => {
-    let clone = Object.assign({}, binding)
-    clone.icon = "ShareAndroid"
-    return clone
+  const typeToField = Object.values(FIELDS).reduce((acc, field) => {
+    acc[field.type] = field
+    return acc
+  }, {})
+
+  const memoStore = memo({
+    row,
+    meta,
   })
 
   let table
+  // Row Schema Fields
   let schemaFields
   let attachmentTypes = [
     FieldType.ATTACHMENTS,
     FieldType.ATTACHMENT_SINGLE,
     FieldType.SIGNATURE_SINGLE,
   ]
 
-  $: {
-    table = $tables.list.find(table => table._id === value?.tableId)
+  let customPopover
+  let popoverAnchor
+  let editableRow = {}
+  let editableFields = {}
 
-    // Just sorting attachment types to the bottom here for a cleaner UX
-    schemaFields = Object.entries(table?.schema ?? {}).sort(
-      ([, schemaA], [, schemaB]) =>
-        (schemaA.type === "attachment") - (schemaB.type === "attachment")
-    )
+  // Avoid unnecessary updates
+  $: memoStore.set({
+    row,
+    meta,
+  })
+
+  $: parsedBindings = bindings.map(binding => {
+    let clone = Object.assign({}, binding)
+    clone.icon = "ShareAndroid"
+    return clone
+  })
+
+  $: tableId = $memoStore?.row?.tableId
 
-    schemaFields.forEach(([, schema]) => {
-      if (!schema.autocolumn && !value[schema.name]) {
-        value[schema.name] = ""
+  $: initData(tableId, $memoStore?.meta?.fields, $memoStore?.row)
+
+  const initData = (tableId, metaFields, row) => {
+    if (!tableId) {
+      return
+    }
+
+    // Refesh the editable fields
+    editableFields = cloneDeep(metaFields || {})
+
+    // Refresh all the row data
+    editableRow = cloneDeep(row || {})
+
+    table = $tables.list.find(table => table._id === tableId)
+
+    if (table) {
+      editableRow["tableId"] = tableId
+
+      schemaFields = Object.entries(table?.schema ?? {})
+        .filter(entry => {
+          const [, field] = entry
+          return field.type !== "formula" && !field.autocolumn
+        })
+        .sort(([nameA], [nameB]) => {
+          return nameA < nameB ? -1 : 1
+        })
+
+      // Parse out any data not in the schema.
+      for (const column in editableFields) {
+        if (!Object.hasOwn(table?.schema, column)) {
+          delete editableFields[column]
+        }
       }
-    })
-  }
-  const onChangeTable = e => {
-    value["tableId"] = e.detail
-    dispatch("change", value)
+      editableFields = editableFields
```

Comment:
Very NAB, feel free to ignore: Is the assignment here necessary to trigger an update? 

If so, since we're cloning the `editableFields` anyway, would it be a little nicer to return a `newEditableFields` from this function and change the reactive statement that invokes it to `$: editableFields = initData(tableId, $memoStore?.meta?.fields, $memoStore?.row)`?

Reply:
I think this line might be a hangover from an earlier iteration. I don't believe this is necessary anymore. 

Also, as `editableFields` is directly altered elsewhere in the same component, I usually prefer to use a standard variable declaration in those scenarios. I find directly altering a reactive variable quite odd.